ibis/sql/mssql/client.py

```diff
@@ -0,0 +1,196 @@
+import contextlib
+import getpass
+
+
+import pyodbc  # NOQA fail early if the driver is missing
+import sqlalchemy as sa
+from sqlalchemy.dialects.mssql.pyodbc import MSDialect_pyodbc
+
+import ibis.expr.datatypes as dt
+import ibis.sql.alchemy as alch
+from ibis.sql.mssql.compiler import MSSQLDialect
+
+
+@dt.dtype.register(MSDialect_pyodbc, sa.dialects.mssql.UNIQUEIDENTIFIER)
+def sa_string(_, satype, nullable=True):
+    return dt.String(nullable=nullable)
+
+
+@dt.dtype.register(MSDialect_pyodbc, sa.dialects.mssql.BIT)
+def sa_boolean(_, satype, nullable=True):
+    return dt.Boolean(nullable=nullable)
+
+
+class MSSQLTable(alch.AlchemyTable):
+    pass
+
+
+class MSSQLSchema(alch.AlchemyDatabaseSchema):
+    pass
+
+
+class MSSQLDatabase(alch.AlchemyDatabase):
+    schema_class = MSSQLSchema
+
+
+class MSSQLClient(alch.AlchemyClient):
+
+    """The Ibis MSSQL client class
+
+    Attributes
+    ----------
+    con : sqlalchemy.engine.Engine
+    """
+
+    dialect = MSSQLDialect
+    database_class = MSSQLDatabase
+    table_class = MSSQLTable
+
+    def __init__(
+        self,
+        host='localhost',
+        user=None,
+        password=None,
+        port=1433,
+        database='mssql',
+        url=None,
+        driver='pyodbc',
+    ):
+        if url is None:
+            if driver != 'pyodbc':
+                raise NotImplementedError(
+                    'pyodbc is currently the only supported driver'
+                )
+            user = user or getpass.getuser()
+            url = sa.engine.url.URL(
+                'mssql+pyodbc',
+                host=host,
+                port=port,
+                username=user,
+                password=password,
+                database=database,
+            )
+        else:
+            url = sa.engine.url.make_url(url)
+        super().__init__(sa.create_engine(url))
+        self.database_name = url.database
+
+    @contextlib.contextmanager
+    def begin(self):
+        with super().begin() as bind:
+            # set timezone utc
+            yield bind
+            # set timezone previous timezone
+
+    def database(self, name=None):
+        """Connect to a database called `name`.
+
+        Parameters
+        ----------
+        name : str, optional
```

Comment:
As we are using `string`, replace `str` to `string`.
probably we need to create a document and define standards, 
but for now I think would be good to use `string`

Reply:
Fixed.